truss/remote/baseten/remote.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+
+from truss.remote.baseten.api import BasetenApi
+from truss.remote.baseten.auth import AuthService
+from truss.remote.baseten.core import (
+    archive_truss,
+    create_model,
+    exists_model,
+    upload_model,
+)
+from truss.remote.baseten.service import BasetenService
+from truss.remote.baseten.utils.transfer import base64_encoded_json_str
+from truss.remote.truss_remote import TrussRemote
+from truss.truss_handle import TrussHandle
+
+
+class BasetenRemote(TrussRemote):
+    def __init__(self, remote_url: str, api_key: Optional[str] = None, **kwargs):
+        super().__init__(remote_url, **kwargs)
+        if api_key:
+            self._auth_service = AuthService(api_key=api_key)
+        else:
+            self._auth_service = AuthService()
+
+    def authenticate(self):
+        return self._auth_service.validate()
+
+    def push(self, truss_handle: TrussHandle, model_name: str):  # type: ignore
+        self.authenticate()
+        api = BasetenApi(f"{self._remote_url}/graphql/", self._auth_service)
+
+        if model_name.isspace():
+            raise ValueError("Model name cannot be empty")
+
+        if exists_model(api, model_name):
```

Comment:
Add a TODO for @joostinyi  to overwrie this when patching incompatible and the user confirms to overwrite. Not for this PR, but just need to not forget how we wanna iterate on these things.